walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,21 +26,41 @@
     //no attendees -- first test
     if (request.getAttendees() == null){
         availibility = Arrays.asList(TimeRange.WHOLE_DAY);
+       
+        //return the collection
+        return availibility;
     }
 
     //no options too long of a request (how do I access time range here? THis is an error) -- second test
     else if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
         availibility = Arrays.asList();
+        
+        //return the collection
+        return availibility;
     }
 
     //event splits day into two options (before and after) -- third test
     for (Event event : events){
         availibility =
-        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen()[0], false),
-            TimeRange.fromStartEnd(event.getWhen()[1], TimeRange.END_OF_DAY, true));
-        
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false),
+            TimeRange.fromStartEnd(event.getWhen().end(), TimeRange.END_OF_DAY, true));
     }
 
+    //every attendee considered -- fourth test
+    for (String attendee : request.getAttendees()){
```

Comment:
I'm trying to do the "consider every attendee" test pass and I'm just a bit stuck on how to iterate to get the times I want for each person's event (to replace the TIME_...)

Reply:
Lines 50 to 53 are a good start.  On there you're doing:
50.  Look at all requested attendees
51.  Look at all existing events
52.  Get attendees for the existing event
53.  Check if the attendee you're looking at (from line #50) is an attendee of the event you're looking at (from line #51)

What then should you do once you find out your requested attendee is already booked for an event?  Before you put code together, try to describe how you would do this with words/sentences.  Try to reply back with a comment with step-by-step descriptions

Reply:
Well, I guess I don't even need specific attendees, just events. I recently did another commit where I did the following:
1. Loop through event1 in events
2. within this loop, loop through event2 in events
3. if event1 isn't the same as event2 and event1 and event2 have no overlap, set the availability to when both parties are not attending another event.

Reply:
Yep it could be helpful at the beginning to think about how to work with the events first, without thinking about specific attendees.  Later on though, you'll need to also consider attendees.  Like this use case for example:
```
// Events  :       |--A--|     |--B--|
// Day     : |-----------------------------|
// Request:  1 hour for Person C
// Options : |----------1------------------|
```


> 1. Loop through event1 in events
> 2.  within this loop, loop through event2 in events
> 3.  if event1 isn't the same as event2 and event1 and event2 have no overlap

This part sounds ok.  I'm not sure I understand "set the availability to when both parties are not attending another event"